apps/fabric-website/src/components/IconGrid/IconGrid.tsx

```diff
@@ -56,36 +58,84 @@ export class IconGrid extends React.Component<IIconGridProps, IIconGridState> {
     );
   }
 
+  public async componentDidMount() {
+    // Load the icons async after mount to avoid them being bundled and parsed with the page,
+    // due to their large bundle size.
+    switch (this.props.iconType) {
+      case 'font':
+        // The explicit json-loader! import helps prevent this from being processed by TS
+        // (we want only webpack to handle it)
+        const fontIcons = (await import('json-loader!@uifabric/icons/lib/data/AllIconNames.json')) as IIconInfo[];
+        // There's also a metadata entry in this file which doesn't have a name
+        this.setState({ icons: fontIcons.filter(icon => !!icon.name) });
+        break;
+
+      case 'core':
+        const coreIcons = (await import('json-loader!office-ui-fabric-core/src/data/icons.json')) as IIconInfo[];
+        this.setState({ icons: coreIcons });
+        break;
+
+      case 'svg':
+        const MDL2Icons = await import('@fluentui/react-icons');
+        this.setState({
+          icons: Object.keys(MDL2Icons)
+            .map((name): IIconInfo | undefined => {
+              const IconComponent = MDL2Icons[name];
+              if (typeof IconComponent === 'function' && /^[A-Z].*Icon$/.test(name)) {
+                return { name, icon: IconComponent };
+              }
+              return undefined;
+            })
+            .filter(icon => !!icon),
+        });
+        break;
+    }
+  }
+
   private _getItems = (): { name: string }[] => {
-    const { icons } = this.props;
-    const { searchQuery } = this.state;
+    const { icons, searchQuery } = this.state;
+    const query = searchQuery.toLowerCase();
 
-    return icons.filter(icon => icon && icon.name && icon.name.toLowerCase().indexOf(searchQuery.toLowerCase()) !== -1);
+    return icons ? (query ? icons.filter(icon => icon.name?.toLowerCase().indexOf(query) !== -1) : icons) : [];
```

Comment:
I changed this to skip the filtering logic unless there's actually a search query

Reply:
smart